Cargo.toml

```diff
@@ -46,6 +46,12 @@ brotli = ["async-compression", "async-compression/brotli"]
 json = ["serde_json"]
 
 trust-dns = ["trust-dns-resolver"]
+trust-dns-over-tls = []
```

Comment:
Oh wow, I didn't consider we'd need to enable these features to use them inside reqwest. All of these new features in reqwest make me a little nervous, is it fair to start conservatively and add configuration that doesn't require adding new features?

Reply:
New features are needed to enable DNS-over-TLS and DNS-over-HTTPS in `trust-dns-resolver`: https://docs.rs/trust-dns-resolver/0.19.5/trust_dns_resolver/#dns-over-tls-and-dns-over-https.

By default (when no features of `trust-dns-resolver` are enabled), DNS queries and responses are sent in plaintext, which, in some cases, may be undesirable.

Should I remove the support for DNS-over-TLS and DNS-over-HTTPS ?

Reply:
Yea I see now the features are required, and I get why. I just initially didn't make too big of a surface area change. I suppose adding new cargo features isn't that different from adding new methods...